cis194/week7/daniel-deng/src/CIS194/Homework07/JoinList.hs

```diff
@@ -0,0 +1,26 @@
+module CIS194.Homework07.JoinList ( JoinList(..) ) where
+
+import CIS194.Homework07.Sized ( Sized(..) )
+
+data JoinList m a = Empty
+                  | Single m a
+                  | Append m (JoinList m a) (JoinList m a)
+  deriving (Eq, Show)
+
+tag :: Monoid m => JoinList m a -> m
```

Comment:
@shineli1984 any way we could derive the monoidal `m` for `JoinList`? For `m` is the only thing in the type constructor that has a `Monoid` constraint.

Reply:
It seems we should be able to derive this. There must be some generic monoid module already doing this, but it's not in the GHC.